### PR TITLE
Clean up '#if CORE' in PSPrincipal.cs

### PR DESCRIPTION
**Pull request by iSazonov**
Related #3565 and #4357.

~~The build is failed with "'TimeZone' is obsolete: 'System.TimeZone has been deprecated."~~

~~It seems 'TimeZone' is necessary for backward compatibility with Windows PowerShell. How can we resolve this? Remove?~~

Some files depends on depredicated TimeZone type - migrate them to use TimeZoneInfo.

### CODE REVIEW

File: src/System.Management.Automation/engine/remoting/fanin/PSPrincipal.cs [hunk 1]
```diff
@@ -126,17 +125,15 @@ public PSSenderInfo(PSPrincipal userPrincipal, string httpUrl)
...
         /// </summary>
-        public TimeZone ClientTimeZone
+        public TimeZoneInfo ClientTimeZone
         {
             get { return clientTimeZone; }
             internal set { clientTimeZone = value; }
         }
-        private TimeZone clientTimeZone;
-#endif
+        private TimeZoneInfo clientTimeZone;
```
Comment (PaulHigin) [inline src/System.Management.Automation/engine/remoting/fanin/PSPrincipal.cs]: Minor issue.  We are trying to move away from explicit field members for properties and instead to this:
```powershell
public TimeZoneInfo ClientTimeZone
{
    get;
    internal set;
}
```
Can you please remove the private clientTimeZone field?

**Review by PaulHigin — CHANGES_REQUESTED** [review]

File: src/System.Management.Automation/engine/remoting/fanin/PSPrincipal.cs (hunk 1 above)
Comment (iSazonov) [inline src/System.Management.Automation/engine/remoting/fanin/PSPrincipal.cs]: Fixed.

**Review by PaulHigin — APPROVED** [review]
LGTM.  Thanks!

File: src/System.Management.Automation/engine/remoting/server/serverremotesession.cs [hunk 2]
```diff
@@ -743,9 +743,8 @@ private void HandleCreateRunspacePool(object sender, RemoteDataEventArgs createR
             // This is used by the initial session state configuration providers like Exchange.
             if (Context != null)
             {
-#if !CORECLR // TimeZone Not In CoreCLR
+
```
Comment (adityapatwardhan) [inline src/System.Management.Automation/engine/remoting/server/serverremotesession.cs]: Please remove extra line.

**Review by adityapatwardhan — CHANGES_REQUESTED** [review]

File: src/System.Management.Automation/engine/remoting/server/serverremotesession.cs (hunk 2 above)
Comment (iSazonov) [inline src/System.Management.Automation/engine/remoting/server/serverremotesession.cs]: Fixed.

**Review by adityapatwardhan — APPROVED** [review]

**Pull request merged**